packages/vector_graphics/lib/src/listener.dart

```diff
@@ -0,0 +1,131 @@
+import 'dart:ui' as ui;
+import 'dart:typed_data';
+
+import 'package:vector_graphics_codec/vector_graphics_codec.dart';
+
+/// A listener implementation for the vector graphics codec that converts the
+/// format into a [ui.Picture].
+class FlutterVectorGraphicsListener extends VectorGraphicsCodecListener {
+
+  /// Create a new [FlutterVectorGraphicsListener].
+  factory FlutterVectorGraphicsListener() {
+    final ui.PictureRecorder recorder = ui.PictureRecorder();
+    final ui.Canvas canvas = ui.Canvas(recorder);
+    return FlutterVectorGraphicsListener._(canvas, recorder);
+  }
+
+  FlutterVectorGraphicsListener._(this._canvas, this._recorder);
+
+  final ui.PictureRecorder _recorder;
+  final ui.Canvas _canvas;
+  final List<ui.Paint> _paints = <ui.Paint>[];
+  final List<ui.Path> _paths = <ui.Path>[];
+  ui.Path? _currentPath;
```

Comment:
Path has a reset method. It's probably faster than allocating a new path.

Reply:
not sure how we would use this. The paths are encoded before the drawPathCommands